lib/matplotlib/colors.py

```diff
@@ -742,6 +742,107 @@ def func_r(x):
 
         return LinearSegmentedColormap(name, data_r, self.N, self._gamma)
 
+    def join(self, other, name=None, frac_self=None, N=None):
```

Comment:
It might be a good idea to also add add a `__add__` method so you can do `cm1 + cm2` and get a 50/50 map?  Could be too cute for it's own good.